subprojects/provider/src/test/kotlin/org/gradle/kotlin/dsl/caching/LoadDirectoryTest.kt

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.kotlin.dsl.caching
+
+import org.gradle.internal.id.UniqueId
+
+import org.gradle.kotlin.dsl.cache.LoadDirectory
+import org.gradle.kotlin.dsl.cache.PackMetadata
+import org.gradle.kotlin.dsl.cache.ScriptBuildCacheKey
+
+import org.gradle.kotlin.dsl.fixtures.TestWithTempFiles
+
+import org.junit.Test
+
+import java.io.InputStream
+
+
+class LoadDirectoryTest : TestWithTempFiles() {
+
+    @Test
+    fun `returns correct OriginMetadata based on buildInvocationId`() {
+
+        // given:
+        val previousBuildInvocation = UniqueId.generate()
+        val currentBuildInvocation = UniqueId.generate()
+        val subject = LoadDirectory(
+            newFolder(),
+            ScriptBuildCacheKey("test-key", "test-key/42"),
+            currentBuildInvocation
+        )
+
+        // then:
+        assert(
+            subject.load(packProducedBy(currentBuildInvocation))!!.metadata.isProducedByCurrentBuild
+        )
+
+        // and:
+        assert(
+            !subject.load(packProducedBy(previousBuildInvocation))!!.metadata.isProducedByCurrentBuild
+        )
```

Comment:
Why not use `assertFalse`?

Reply:
For symmetry.

Reply:
😕 Symmetry? Aren't `assertFalse` and `assertTrue` pretty standard?

Reply:
The code is not using `assertTrue`.

Reply:
Moved to `assertFalse` / `assertTrue`.